### PR TITLE
Cookies and headers

### PR DESCRIPTION
**Pull request by peterlynch**
port of https://github.com/sonatype/nexus-oss/commit/f9ed01dea4eed079669c36e64bed72f7c67ec9fc

http://bamboo.s/browse/NX-OSSF504-2 :+1: 

made the Nexus context path configurable via NexusBundleConfiguration
added getSecureUrl to go with enableSsl() option.

NEXUS-7878 - Restlet header duplication removal

NEXUS-7879 - make Secure flag on session Set-Cookie value dynamic based on secure request instead of configurable

NEXUS-5830 - prevent status resource from creating a session and returning Set-Cookie header

NEXUS-7882 - make logout return a Set-Cookie header as expected that forces delete/expiry of the session cookie

NEXUS-7880 - make Nexus cookie name configurable and change the default name

add robustness to modern session Cookie ITs and eliminate legacy cookie ITs

### CODE REVIEW

**Comment by peterlynch** [comment]
/service/local/logout does return NXSESSIONID=deleteMe , but UI does not use this to logout.

UI is using /service/extdirect resource to sign out.

Not clear if I should write IT testing extdirect signout resource in my cookie IT or is there another IT to write using some other frameworky thing?

**Comment by peterlynch** [comment]
now that I added a test for /service/extdirect logout request, I see the Secure flag is not coming back on the logout NXSESSIONID cookie. Adding WIP tag back until this is resolved.

File: components/nexus-core/src/test/java/org/sonatype/nexus/web/CookieFilterTest.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+import org.mockito.Mock;
+
+import static java.util.Arrays.asList;
+import static org.mockito.Mockito.anyString;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyZeroInteractions;
+import static org.mockito.Mockito.when;
+
+public class CookieFilterTest
```
Comment (jdillon) [inline components/nexus-core/src/test/java/org/sonatype/nexus/web/CookieFilterTest.java]: minor... anal

```
/**
 * Tests for {@link CookieFilter}.
 */
```

**Comment by jdillon** [comment]
+1

File: components/nexus-core/src/main/java/org/sonatype/nexus/web/CookieFilter.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+{
+  private static final String SECURE_FLAG = "; Secure";
+
+  @Override
+  protected boolean preHandle(final ServletRequest request, final ServletResponse response) throws Exception {
+    filterCookies(request, response);
+    return true;
+  }
+
+  @Override protected void postHandle(final ServletRequest request, final ServletResponse response) throws Exception {
```
Comment (peterlynch) [inline components/nexus-core/src/main/java/org/sonatype/nexus/web/CookieFilter.java]: for some reason IDEA is happy with either annotation on new line or same line :(

**Comment by mrprescott** [comment]
+1

**Comment by peterlynch** [comment]
manually squashed and merged: https://github.com/sonatype/nexus-oss/commit/059b0843f10d4417a9029b8d680f930c8461338e

**Pull request closed without merging**